docs/03_goclient/15_sessions.md

```diff
@@ -0,0 +1,134 @@
+  ao := workflow.ActivityOptions{
+    ScheduleToStartTimeout: time.Second * 5,
+    StartToCloseTimeout:    time.Minute,
+  }
+  ctx = workflow.WithActivityOptions(ctx, ao)
+
+  so := &workflow.SessionOptions{
+    CreationTimeout:  time.Minute,
+    ExecutionTimeout: time.Minute,
+  }
+  sessionCtx, err := workflow.CreateSession(ctx, so)
+  if err != nil {
+    return err
+  }
+  defer workflow.CompleteSession(sessionCtx)
+
+  var fInfo *fileInfo
+  err = workflow.ExecuteActivity(sessionCtx, downloadFileActivityName, fileID).Get(sessionCtx, &fInfo)
+  if err != nil {
+    return err
+  }
+
+  var fInfoProcessed *fileInfo
+  err = workflow.ExecuteActivity(sessionCtx, processFileActivityName, *fInfo).Get(ctx, &fInfoProcessed)
+  if err != nil {
+    return err
+  }
+
+  return workflow.ExecuteActivity(sessionCtx, uploadFileActivityName, *fInfoProcessed).Get(ctx, nil)
+}
+```
+
+## Session Metadata
+
+```go
+type SessionInfo struct {
+  // A unique ID for the session
+  SessionID         string
+
+  // The hostname of the worker that is executing the session
+  HostName          string
+
+  // ... other unexported fields
+}
+
+func GetSessionInfo(ctx Context) *SessionInfo
+```
+
+The session context also stores some metadata of the session, which can be retrieved by the `GetSessionInfo()` API. For now, there are only two exported fields in the returned `workflow.SessionInfo`. We may consider exposing more information in the future.
+
+## Concurrent Session Limitation
+
+It's very easy to limit the number of concurrent sessions running on a worker, just set the `MaxConcurrentSessionExecutionSize` field of `worker.Options` to the desired value. By default this field is set to a very large value, so there's no need to manually set it if no limitation is needed.
+
+If a worker hits this limitation, it won't accept any new `CreateSession()` request until one of the existing sessions is completed and `CreateSession()` will return an error if the session can't be created within `CreationTimeout`.
+
+## Recreate Session
+
+For long-running sessions, user may want to use the ContinueAsNew feature to split the workflow into multiple runs, but still needs all the activities to be executed by the same worker. The `RecreateSession()`  API is designed for such use case.
+
+```go
+func RecreateSession(ctx Context, recreateToken []byte, sessionOptions *SessionOptions) (Context, error)
+```
+
+Its usage is the same as `CreateSession()` except that it also takes in a `recrateToken`, which is needed to create a new session on the same worker as the previous one. User can get the token by calling the `GetRecreateToken()` method of the `SessionInfo` object and pass it to the next run.
+
+```go
+token := workflow.GetSessionInfo(sessionCtx).GetRecrateToken()
+```
+
+## Q&A
+
+### Is there a complete sample code?
+Yes, the fileprocessing example in the cadence-sample repo has been updated to use the session framework.
```

Comment:
Add a link to the sample repo. Even better, link directly to the file processing example.